node/src/reactor/joiner.rs

```diff
@@ -932,6 +932,7 @@ impl Reactor {
     /// the network, closing all incoming and outgoing connections, and frees up the listening
     /// socket.
     pub async fn into_validator_config(self) -> Result<ValidatorInitConfig, Error> {
+        let latest_block = self.linear_chain_sync.latest_block().cloned();
```

Comment:
Can't we read it from the storage?

Reply:
I'm not sure how at this point. We'd probably have to pass an `effect_builder` as an additional parameter and call `get_highest_block_from_storage` (at least this is an `async` function, so we could `await` the result directly).
But we keep track of the latest block, anyway, so why not use it?

Reply:
There's a synchronous API there as well. You can add a method you need and call it in `validator.rs` when constructing the reactor's components.

Reply:
I think it makes sense not to read the data again, if we have it, anyway, after completing the syncing process. So since you approved, I'll merge it as is for now :wink: